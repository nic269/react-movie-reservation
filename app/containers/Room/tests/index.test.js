/**
 *
 * Tests for Room
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';
import { Room } from '../index';

jest.mock('components/SeatLayout', () => 'mockedSeatLayout');
jest.mock('lodash/debounce', () => jest.fn(fn => fn));

describe('<Room />', () => {
  let store;
  let props;

  beforeAll(() => {
    props = {
      onSelect: jest.fn(),
    };
    store = configureStore({}, browserHistory);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Room {...props} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('zoom in seat layout', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Room {...props} />
      </Provider>,
    );
    fireEvent.click(getByTestId('zoom-in'));
    expect(getByTestId('zoom-level')).toHaveTextContent('2x');
    fireEvent.click(getByTestId('zoom-in'));
    expect(getByTestId('zoom-level')).toHaveTextContent('3x');
    fireEvent.click(getByTestId('zoom-in'));
    expect(getByTestId('zoom-level')).toHaveTextContent('3x');
  });

  it('zoom out seat layout', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Room {...props} />
      </Provider>,
    );
    fireEvent.click(getByTestId('zoom-in'));
    expect(getByTestId('zoom-level')).toHaveTextContent('2x');
    fireEvent.click(getByTestId('zoom-out'));
    expect(getByTestId('zoom-level')).toHaveTextContent('1x');
    fireEvent.click(getByTestId('zoom-out'));
    expect(getByTestId('zoom-level')).toHaveTextContent('1x');
  });
});
