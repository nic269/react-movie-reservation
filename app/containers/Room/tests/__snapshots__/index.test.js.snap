// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Room /> Should render and match the snapshot 1`] = `
<div>
  <button
    data-testid="zoom-in"
    type="button"
  >
    +
  </button>
  <span
    data-testid="zoom-level"
  >
    1
    x
  </span>
  <button
    data-testid="zoom-out"
    type="button"
  >
    -
  </button>
</div>
`;
